Posts/2021/02/XML-NewEmployees.md

```diff
@@ -0,0 +1,175 @@
+```yaml
+---
+post_title: 'Leveraging XML with New Employee PowerShell Scripts'
+username: s-r-turner
+categories: PowerShell
+tags: XML, ActiveDirectory
+featured_image:
+summary: using XML referencing to bolster up your new employee scripts with information.
+---
+```
+
+# Leveraging XML With New Employee PowerShell Scripts
+
+I'm going to show how you can leverage XML files for referencing information, to help bolster up your Active Directory user accounts. 
+This will ensure things such as Outlook contact cards are correct but also allows you to use this information at a later date, e.g. creating dynamic distribution lists based on office locations, or setting NTFS permissions on fire evacuation plans for specific floors in buildings.
+Using XML allows for non-PowerShell savvy users to jump into the files and amend or add address information. Meaning less plain text strings in scripts.
+
+## Scenario
+
+You have multiple subsidiary companies that each have their own individual sets of offices and addresses. 
+You need to ensure that each new user you create has the correct address and contact information based on where they are working.
+Whilst in this post I am talking about User Accounts, there is no reason why you can't apply the same logic for Computer Accounts or Resources. 
+No one likes asking around for the location of a meeting room you were supposed to be in 10 minutes ago!
+
+## Getting Started - Creating the XML
+
+To create the xml, we're going to use the [XmlWriter](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlwriter?view=net-5.0) class, put this into a variable to build upon and add some formatting options (this makes it easier to see groups of information). Like so:
```

Comment:
Always remove the `?view=<version>` and the locale `en-us/` from URL on docs.microsoft.com.

```suggestion
To create the xml, we're going to use the [XmlWriter](https://docs.microsoft.com/dotnet/api/system.xml.xmlwriter) class, put this into a variable to build upon and add some formatting options (this makes it easier to see groups of information). Like so:
```